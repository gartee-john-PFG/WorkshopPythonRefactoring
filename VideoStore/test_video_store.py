# from approvaltests.approvals import verify
from approvaltests.combination_approvals import verify_all_combinations
from customer import Customer, Movie, Rental

def test_what_does_customer_statement_do():
    customer_name = ["Test Customer"]
    movie_name = ["Star Wars"]
    price_code = [Movie.REGULAR, Movie.CHILDRENS, Movie.NEW_RELEASE]
    days_rented = [2, 3, 4]

    # result = make_statement(customer_name, movie_name, price_code)
    
    # verify(result)
    verify_all_combinations(make_statement,[customer_name, movie_name, price_code, days_rented])

def make_statement(customer_name, movie_name, price_code, days_rented):
    cust = Customer(customer_name)
    movie = Movie(movie_name, price_code)
    rental = Rental(movie, days_rented)
    cust.add_rental(rental)

    result = cust.statement()
    return result