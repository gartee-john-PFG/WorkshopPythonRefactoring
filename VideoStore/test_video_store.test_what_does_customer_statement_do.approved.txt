args: ('Test Customer', 'Star Wars', 0, 2) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 2\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 0, 3) => 'Rental Record for Test Customer\n\tStar Wars\t1.5\nAmount owed is 3.5\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 0, 4) => 'Rental Record for Test Customer\n\tStar Wars\t3.0\nAmount owed is 5.0\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 2) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 1.5\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 3) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 1.5\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 2, 4) => 'Rental Record for Test Customer\n\tStar Wars\t1.5\nAmount owed is 3.0\nYou earned 1 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 2) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 6\nYou earned 2 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 3) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 9\nYou earned 2 frequent renter points'
args: ('Test Customer', 'Star Wars', 1, 4) => 'Rental Record for Test Customer\n\tStar Wars\t0\nAmount owed is 12\nYou earned 2 frequent renter points'
